### PR TITLE
extract version logic to config package

### PR DESCRIPTION
**Pull request by mscurtescu**
- extract version login to a new `version.go` file in the `config` package
- add tests for version functions
- generate version string based on byte array
- remove hard coded instances of version string

### CODE REVIEW

**Comment by mscurtescu** [comment]
@CassOnMars I looked into adding the version to the output of `GetNodeInfo` rpc, but I don't know how to regenerate the Go clients for protobuf. I can run the `Makefile`, have `protoc` installed, including the Go plugin, but it still fails:
```
protoc -I=. --go-grpc_out=paths=source_relative:. \
	--grpc-gateway_out . \
    --grpc-gateway_opt logtostderr=true \
    --grpc-gateway_opt paths=source_relative \
    --grpc-gateway_opt generate_unbound_methods=true \
	--go_out=paths=source_relative:. *.proto
protoc-gen-grpc-gateway: program not found or is not executable
Please specify a program using absolute path or make sure the program is available in your PATH system variable
--grpc-gateway_out: protoc-gen-grpc-gateway: Plugin failed with status code 1.
make: *** [Makefile:8: node.pb.go] Error 1
```

Any suggestions? Not very important, just curious.

**Comment by CassOnMars** [comment]
> I looked into adding the version to the output of GetNodeInfo rpc, but I don't know how to regenerate the Go clients for protobuf. I can run the Makefile, have protoc installed, including the Go plugin, but it still fails:

you need the protoc-gen-go-grpc tool, this adds both:

```
go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28
go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2
```

File: docker-compose.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ services:
       retries: 3
       start_period: 1m
     volumes:
-      - ./.config:/opt/ceremonyclient/node/.config
+      - ./.config:/root/.config
```
Comment (CassOnMars) [inline docker-compose.yml:34]: will this cause a problem for anyone currently running this?

Comment (mscurtescu) [reply docker-compose.yml:34]: they will not see the the `.config` folder on the host, the container will run fine, so they cannot easily backup `config.yml` and `keys.yml`

**Comment by mscurtescu** [comment]
> > I looked into adding the version to the output of GetNodeInfo rpc, but I don't know how to regenerate the Go clients for protobuf. I can run the Makefile, have protoc installed, including the Go plugin, but it still fails:
> 
> you need the protoc-gen-go-grpc tool, this adds both:
> 
> ```
> go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28
> go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2
> ```

Still the same error (I already had both installed, maybe not the exact version for the second one):
```shell
$ make
protoc -I=. --go-grpc_out=paths=source_relative:. \
	--grpc-gateway_out . \
    --grpc-gateway_opt logtostderr=true \
    --grpc-gateway_opt paths=source_relative \
    --grpc-gateway_opt generate_unbound_methods=true \
	--go_out=paths=source_relative:. *.proto
protoc-gen-grpc-gateway: program not found or is not executable
Please specify a program using absolute path or make sure the program is available in your PATH system variable
--grpc-gateway_out: protoc-gen-grpc-gateway: Plugin failed with status code 1.
make: *** [Makefile:8: node.pb.go] Error 1
```

The `protoc`, `protoc-gen-go` and `protoc-gen-go-grpc` binaries are available, but not `protoc-gen-grpc-gateway`.

We can follow up on this one in a future PR, let's get the version normalisation in.

File: docker-compose.yml (hunk 1 above)
Comment (mcloebel) [reply docker-compose.yml:34]: Thanks for fixing this.

**Pull request merged**